### PR TITLE
[SPARK-43959][SQL][TESTS] Make RowLevelOperationSuiteBase and AlignAssignmentsSuite abstract

### PR DESCRIPTION
**Pull request by aokolnychyi**
### What changes were proposed in this pull request?

This PR makes `RowLevelOperationSuiteBase` and `AlignAssignmentsSuite` abstract.

### Why are the changes needed?

These changes are needed to address [this](https://github.com/apache/spark/pull/41028#discussion_r1190595232) comment.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
cc @cloud-fan @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (Pending CIs).

**Review by HyukjinKwon — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlignAssignmentsSuite.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ import org.apache.spark.sql.execution.datasources.v2.V2SessionCatalog
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types.{BooleanType, IntegerType, StructType}
 
-class AlignAssignmentsSuite extends AnalysisTest {
+abstract class AlignAssignmentsSuite extends AnalysisTest {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlignAssignmentsSuite.scala:39]: nit: if this is a base suite, probably add a `Base` suffix in the name, to be consistent with `RowLevelOperationSuiteBase`

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlignAssignmentsSuite.scala (hunk 1 above)
Comment (aokolnychyi) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlignAssignmentsSuite.scala]: Agree, updated.

**Comment by dongjoon-hyun** [comment]
Could you re-trigger once more, @aokolnychyi ? The required pipeline seems to be hung and hit timeout.

```
2023-06-06T04:37:18.0612857Z [0m[[0m[0minfo[0m] [0m[0m[34mTest run [0mtest.org.apache.spark.sql.[33mJavaRowSuite[0m[34m finished: [0m[34m0 failed[0m[34m, [0m[34m0 ignored[0m[34m, 2 total, 0.002s[0m[0m
2023-06-06T08:15:03.6779310Z 
2023-06-06T08:15:05.6773971Z Session terminated, killing shell... ...killed.
2023-06-06T08:15:05.7111801Z ##[error]The operation was canceled.
```

**Comment by aokolnychyi** [comment]
Triggered again, let's see [this](https://github.com/aokolnychyi/spark/actions/runs/5172839102/jobs/9355353455) run.

**Comment by dongjoon-hyun** [comment]
It fails again 14 minutes ago. It seems to fail consistently for some reasons. Let me check the `master` branch status.
Screenshot 2023-06-06 at 9 45 54 AM

**Comment by dongjoon-hyun** [comment]
The `master` branch is the same. Let me check this manually and merge. Thank you for your patience.

**Comment by dongjoon-hyun** [comment]
```
$ build/sbt "sql/testOnly org.apache.spark.sql.connector.*"
...
[info] Run completed in 5 minutes, 10 seconds.
[info] Total number of tests run: 809
[info] Suites: completed 24, aborted 0
[info] Tests: succeeded 809, failed 0, canceled 0, ignored 1, pending 0
[info] All tests passed.
[success] Total time: 508 s (08:28), completed Jun 6, 2023 9:59:19 AM
```

```
$ build/sbt "sql/testOnly org.apache.spark.sql.execution.command.*"
...
[info] Run completed in 4 minutes, 32 seconds.
[info] Total number of tests run: 809
[info] Suites: completed 78, aborted 0
[info] Tests: succeeded 809, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 308 s (05:08), completed Jun 6, 2023 10:10:03 AM
```

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you, @aokolnychyi , @cloud-fan , @HyukjinKwon !

**Comment by aokolnychyi** [comment]
Thanks, @dongjoon-hyun @HyukjinKwon @cloud-fan!

**Pull request closed without merging**